src/sysupdate/sysupdate.c

```diff
@@ -751,8 +754,9 @@ static int context_vacuum(
         else
                 log_info("Making room for %" PRIu64 " updates%s", space, special_glyph(SPECIAL_GLYPH_ELLIPSIS));
 
-        for (size_t i = 0; i < c->n_transfers; i++) {
-                r = transfer_vacuum(c->transfers[i], space, extra_protected_version);
+        FOREACH_ARRAY(tr, c->transfers, c->n_transfers) {
+                Transfer *t = *tr;
+                r = transfer_vacuum(t, space, extra_protected_version);
```

Comment:
style nit: add blank line b/t variable declaration and the rest of the code (and everywhere else in this commit - there are some places where you dropped the blank space too, please put it back)